Sources/TestingMacros/Support/DiagnosticMessage.swift

```diff
@@ -189,7 +189,7 @@ struct DiagnosticMessage: SwiftDiagnostics.DiagnosticMessage {
   /// - Returns: A diagnostic message.
   static func genericDeclarationNotSupported(_ decl: some SyntaxProtocol, whenUsing attribute: AttributeSyntax, becauseOf genericClause: some SyntaxProtocol) -> Self {
     Self(
-      syntax: Syntax(genericClause),
+      syntax: (genericClause.root != decl.root) ? Syntax(decl) : Syntax(genericClause),
```

Comment:
Nodes from the lexical context are detached, so diagnostics applied to them don't align to any real source location. This check is only (currently) needed here since we use this diagnostic unmodified for both code paths and prefer that the diagnostic point to the generic clause rather than the affected symbol.